sontruongces/liferay-portal
# LPS 172261 | master

Motivation
=======
This work solves the problem reported in the [LPS-172261](https://issues.liferay.com/browse/LPS-172261) ticket.
cc: @sontruongces @tototrinh 

Proposed Solution
========
Cause:
   - The modified date that displays on the "Documents and Media" UI is fetched from `DLFileVersion`.
   After updating the Permissions of a document, no records will be added to `DLFileVersion`. A new `DLFileVersion` record will be added when the user edits the document (use the 'Edit' action).
   Both actions can make updating the modified date of the document in table `DLFileEntry`.

What we did to solve this problem:
   - Using modified date from `DLFileEntry` for displaying.

Steps to Verify
========
1. Go to Default site > Content & Data > Documents and Media
2. Upload several documents (wait a few minutes between each upload)
3. Update a document permission settings
4. Sort Modified Date + descending-order

- Actual result: The document appears at the top of the list even though the Modified Date has not been updated.
- Expected result: The Modified Date is shown correctly.
